### DISCUSSION TITLE
默认设置第一次运行出错

### DISCUSSION THREAD

**Discussion by yzcj105** [nshen/learn-neovim-lua#144] [Q&A] [answered]
直接git clone 后，第一次使用时报如下错误，系统为mac, neovim v0.8.1 已经试过:TSUpdateSync
求助 

` vim.highlight.create is deprecated, use vim.api.nvim_set_hl instead. See :h deprecated
This function will be removed in Nvim version 0.9
stack traceback:
        ...ar/neovim/0.8.1/share/nvim/runtime/lua/vim/highlight.lua:14: in function 'create'
        ...k/packer/start/toggleterm.nvim/lua/toggleterm/colors.lua:99: in function 'set_highlights'
        ...ite/pack/packer/start/toggleterm.nvim/lua/toggleterm.lua:359: in function 'setup'
        ...s/jingchen/.config/nvim/lua/plugin-config/toggleterm.lua:8: in main chunk
        [C]: in function 'require'
        /Users/jingchen/.vim_configs/new/config_nvim/init.lua:20: in main chunk
Error detected while processing BufReadPost Autocommands for "*":
Error executing lua callback: ...ocal/Cellar/neovim/0.8.1/share/nvim/runtime/filetype.lua:22: Error executing lua: ...ocal/Cellar/neovim
/0.8.1/share/nvim/runtime/filetype.lua:23: Vim(append):Error executing lua callback: .../start/nvim-treesitter/lua/nvim-treesitter/highl
ight.lua:14: attempt to index local 'hlmap' (a nil value)
stack traceback:
        .../start/nvim-treesitter/lua/nvim-treesitter/highlight.lua:14: in main chunk
        [C]: in function 'resolve_module'
        ...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:473: in function 'detach_module'
        ...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:487: in function 'reattach_module'
        ...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:106: in function <...er/start/nvim-treesitter/lua/nvim-treesitter/co
nfigs.lua:105>
        [C]: in function 'nvim_cmd'
        ...ocal/Cellar/neovim/0.8.1/share/nvim/runtime/filetype.lua:23: in function <...ocal/Cellar/neovim/0.8.1/share/nvim/runtime/file
type.lua:22>
        [C]: in function 'nvim_buf_call'
        ...ocal/Cellar/neovim/0.8.1/share/nvim/runtime/filetype.lua:22: in function <...ocal/Cellar/neovim/0.8.1/share/nvim/runtime/file
type.lua:11>
stack traceback:
        [C]: in function 'nvim_cmd'
        ...ocal/Cellar/neovim/0.8.1/share/nvim/runtime/filetype.lua:23: in function <...ocal/Cellar/neovim/0.8.1/share/nvim/runtime/file
type.lua:22>
        [C]: in function 'nvim_buf_call'
        ...ocal/Cellar/neovim/0.8.1/share/nvim/runtime/filetype.lua:22: in function <...ocal/Cellar/neovim/0.8.1/share/nvim/runtime/file
type.lua:11>
stack traceback:
        [C]: in function 'nvim_buf_call'`

**Comment by yzcj105** [comment]
```
stack traceback:
        [C]: in function 'nvim_set_option_value'
        vim/_meta.lua: in function '_set'
        vim/_meta.lua: in function '__newindex'
        .../pack/packer/start/dashboard-nvim/lua/dashboard/init.lua:50: in function 'set_buf_local_options'
        .../pack/packer/start/dashboard-nvim/lua/dashboard/init.lua:392: in function 'instance'
        ...te/pack/packer/start/dashboard-nvim/plugin/dashboard.lua:11: in function <...te/pack/packer/start/dashboard-nvim/plugin/dashb
oard.lua:9>
[packer.nvim] [ERROR 21:12:15] async.lua:20: Error in coroutine: ...ite/pack/packer/start/packer.nvim/lua/packer/display.lua:821: Vim(ap
pend):Error executing lua callback: ...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:473: loop or previous error loading modu
le 'nvim-treesitter.highlight'
[packer.nvim] stack traceback:
[packer.nvim] ^I[C]: in function 'resolve_module'
[packer.nvim] ^I...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:473: in function 'detach_module'
[packer.nvim] ^I...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:487: in function 'reattach_module'
[packer.nvim] ^I...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:106: in function <...er/start/nvim-treesitter/lua/nvim-trees
itter/configs.lua:105>
[packer.nvim] ^I[C]: in function 'resume'
[packer.nvim] ^I.../site/pack/packer/start/packer.nvim/lua/packer/async.lua:12: in function <.../site/pack/packer/start/packer.nvim/lua/
packer/async.lua:11>
```

**Comment by nshen** [comment, marked as answer]
抱歉才看见这个帖子不知道解决没，看着 neovim 版本比较新，是一些插件调用了neovim  deprecated 的 api，有可能是插件还没支持neovim 0.81，也可能是用了旧版本的插件还没支持新API。如果用的 master 分支的代码，可以把这行删掉重新更新插件 https://github.com/nshen/learn-neovim-lua/blob/7af70083eaf469fccb2ad601eaeabac150d080e5/lua/plugins.lua#L154

因为这行会锁定插件版本。

**Reply by yzcj105** [reply]
谢谢！用效果！
